universal/include/userver/http/predefined_header.hpp

```diff
@@ -6,11 +6,17 @@
 #include <fmt/core.h>
 
 #include <userver/utils/trivial_map.hpp>
+#include <userver/utils/small_string.hpp>
 
 USERVER_NAMESPACE_BEGIN
 
 namespace http::headers {
 
+// According to https://www.chromium.org/spdy/spdy-whitepaper/
+// "typical header sizes of 700-800 bytes is common"
+inline constexpr std::size_t kTypicalHeadersSize = 1024;
+using HeadersString = utils::SmallString<kTypicalHeadersSize>;
```

Comment:
This doesn't belong here, i think.

This also bring `small_string.hpp` (which brings boost::) include into too many places, which should slow the compilation down.